pages/[user-id]/my-courses.js

```diff
@@ -2,79 +2,127 @@
+import Link from 'next/link'
 
-import MyCourses from '../../components/MyCourses/MyCourses.js';
-import MyCoursesExampleData from '../../components/MyCourses/data/MyCourses.example.js';
-import { getUserInfo, getCoursesByMenteeId, getCoursesByMentorId } from '../../utils/api/apiCalls.js';
+import {
+  getUserInfo,
+  getCoursesByMenteeId,
+  getCoursesByMentorId,
+  removeCourse,
+  updateCourseInfo,
+  createNewCourse,
+  removeMenteeFromCourse } from '../../utils/api/apiCalls.js';
+import MyCoursesCard from '../../components/MyCourses/MyCoursesCard.js';
+import CreateClassModal from '../../components/CreateClassModal/CreateClassModal.js'
+import MainButton from '../../components/basecomponents/MainButton.js'
+import styles from '../../utils/styles/MyCoursesStyles/MyCourses.module.css';
 
 export default function MyCoursesPage() {
   const router = useRouter();
   var pathUserId = router.asPath.split('/');
   const userID = pathUserId[1];
 
+  const [userInfo, setUserInfo] = useState({ userType: '', id: '', name: { first_name: '', last_name: '' }, avatar_photo: '' })
   const [myCoursesData, setCoursesData] = useState([]);
-  const [userType, setUserType] = useState('');
-
-  const fetchUserInfo = () => {
-    return getUserInfo(userID)
-      .then(res => setUserType(res.account_type))
-      .catch(err => console.log('Error getting user information'))
-  }
 
   const fetchAllCourses = () => {
-    if (userType === 'Mentor') {
+    if (userInfo.userType === 'Mentor') {
       getCoursesByMentorId(userID)
         .then(res => setCoursesData(res))
-        .catch(err => console.log('Error getting course info'))
-    } else if (userType === 'Mentee') {
+        .catch(err => console.warn('Error getting course info'))
+    } else if (userInfo.userType === 'Mentee') {
       getCoursesByMenteeId(userID)
         .then(res => setCoursesData(res))
-        .catch(err => console.log('Error getting course info'))
+        .catch(err => console.warn('Error getting course info'))
+    }
+  }
+
+  const handleDeleteCourse = (courseID) => {
+    removeCourse(courseID)
+      .then(() => fetchAllCourses())
+      .catch(err => console.log('Error deleting course'));
+  }
+
+  const handleStudentDropCourse = (courseID) => {
+    let dropCourseBody = {
+      mentees: {
+        id: userID
+      }
     }
+    removeMenteeFromCourse(courseID, dropCourseBody)
+      .then(() => fetchAllCourses())
+      .catch(err => console.warn('Could not drop course'))
   }
 
-  const handleDeleteCourse = (index) => {
-    let myCoursesCopy = [...myCoursesData];
-    myCoursesCopy.splice(index, 1);
-    setCoursesData(myCoursesCopy);
+  const handleEditCourse = (course, newCourseName, newStartTime, newEndTime) => {
+    let courseUpdateBody = {
+      name: newCourseName,
+      start_time: newStartTime,
+      end_time: newEndTime
+    }
+    updateCourseInfo(course.id, courseUpdateBody)
+      .then(() => fetchAllCourses())
```

Comment:
Same as above, dont need to wrap in arrow if we have no args. Just make sure not to invoke the function if you do that

Reply:
👍